cmd/controller-manager/app/controller-manager.go

```diff
@@ -303,7 +303,12 @@ func setOptionsByFederationConfig(opts *options.Options) {
 	if fedConfig == nil {
 		// FederationConfig could not be sourced from --federation-config or from the API.
 		// create a new `FederationConfig` with default values.
-		fedConfig = &corev1a1.FederationConfig{}
+		fedConfig = &corev1a1.FederationConfig{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      util.FederationConfigName,
+				Namespace: opts.Config.FederationNamespace,
```

Comment:
It seems like `opts.Config.FederationNamespace` will not be correctly set at this point, since it is set in the calling code after this method is called.

Reply:
`opts.Config.FederationNamespace` is set by flag, not code.

Reply:
OK